pkg/util/utils.go

```diff
@@ -134,3 +139,60 @@ func IsDirectory(path string) (bool, error) {
 	}
 	return fileInfo.IsDir(), nil
 }
+
+type ServiceContext struct {
+	Service string `json:"service"`
+	Version string `json:"version"`
+}
+
+type Message struct {
+	Message        string `json:"message"`
+	ServiceContext `json:"serviceContext"`
+}
+
+func ReportError(err error, url string) error {
+	// Extract the stacktrace from the error messages in their orig format
+	if err == nil {
+		return errors.New("Error: ReportError was called with nil error value")
+	}
+	errMsg := fmt.Sprintf("%+v\n", err)
```

Comment:
This would probably be easier to test if you break this function up a bit.

Maybe:
FormatError(error) string
MarshalError(string) byte[]
UploadError(byte[])